ddtrace/profiling/collector/_task.pyx

```diff
@@ -1,25 +1,35 @@
+import sys
 import weakref
 
 from ddtrace.internal import compat
-from ddtrace.internal import nogevent
+from ddtrace.vendor.wrapt.importer import when_imported
 
 from .. import _asyncio
 from .. import _threading
 
 
-try:
-    import gevent.hub
-    import gevent.thread
-    from greenlet import getcurrent
-    from greenlet import greenlet
-    from greenlet import settrace
-except ImportError:
-    _gevent_tracer = None
-else:
+_gevent_tracer = None
+
+
+@when_imported("gevent")
```

Comment:
Should we be using `ModuleWatchdog` for this?

I'd have to look to remind myself, but iirc `@when_imported` doesn't always work.

cc @P403n1x87

Reply:
We could, but this seems to work just fine for now. Perhaps we can refactor if we push for dropping the vendored `wrapt` in favour of our `internal` stuff?

Reply:
what is the impact is this doesn't work properly? we'd be missing gevent task associations with profiles?

Reply:
Yes I think we would miss greenlets and won't collect their stacks

Reply:
ok, I'd say the risk then is fairly low. possibly bad product experience when using gevent, but no ill effects to the application it self 👍🏻